### ISSUE TITLE
ethash stratum_mode stratum makes incorrect handshake

### ISSUE THREAD

**Issue by JKKGBE** [todxx/teamredminer#127]
When using ethash, your miner puts `EthereumStratum/1.0.0` 2nd param into the `params` array, when doing miner-proxy spec, and proceeds to accept ES1 work format, but later submits in miner-proxy fashion.
```
{
  id: 1,
  method: 'mining.subscribe',
  params: [ 'teamredminer/0.7.6', 'EthereumStratum/1.0.0' ]
}
```

And the submission follows, miner-proxy style:
```
{
  id: 3,
  method: 'mining.submit',
  params: 
   [ '0x123456789abcdef123456789abcdef123456789a',
     '1',
     '0xed15126e29532491',
     '0xf31a5182f898728459945b3074183d5d3334e1a3ef32dc714c4d12f434f9a444',
     '0x6e7fa2009b487df4906e4df6b58b60e4c54dc1fa5863d7bf887b4636790059d5' ]
}
```

You have a nicehash mode so I don't see a point to this, it's causing unnecessary trouble.

**Comment by todxx** [comment]
For what is this causing unnecessary trouble?  Is there a specific pool you are having problems with?

**Comment by JKKGBE** [comment]
Yes, ours. I can't tell the details at this point, but it's tipping off the connection system which renders this mode of your miner incompatible and we'd like to support it. The fix would be to just remove it.
